### ISSUE TITLE
Testing new GPTQ models with group-size error.

### ISSUE THREAD

**Issue by BugMonkey42335** [oobabooga/text-generation-webui#577]
### Describe the bug

Trying to generate with newest GPTQ models results in error listed.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

Git pulled latest text-generation-webui & GPTQ-for-LLaMa.

### Logs

```shell
Windows PowerShell
Copyright (C) Microsoft Corporation. All rights reserved.

Try the new cross-platform PowerShell https://aka.ms/pscore6

PS C:\Users\Standard> powershell -ExecutionPolicy ByPass -NoExit -Command "& 'C:\Users\Standard\miniconda3\shell\condabin\conda-hook.ps1' ; conda activate 'C:\Users\Standard\miniconda3' "
(base) PS C:\Users\Standard> conda activate textgen
(textgen) PS C:\Users\Standard> cd text-generation-webui
(textgen) PS C:\Users\Standard\text-generation-webui> python server.py --cai-chat --auto-devices --no-cache --wbits 4 --pre_layer 20 --groupsize 128 --model_type llama
The following models are available:

1. llama-7b-4bit-128g
2. OPT-2.7B-Erebus
3. OPT-350M-Erebus
4. pygmalion-1.3b
5. pygmalion-2.7b
6. pygmalion-350m

Which one do you want to load? 1-6

1

Loading llama-7b-4bit-128g...
Loading model ...
Done.
Loaded the model in 2.47 seconds.
Loading the extension "gallery"... Ok.
C:\Users\Standard\miniconda3\envs\textgen\lib\site-packages\gradio\deprecation.py:40: UserWarning: The 'type' parameter has been deprecated. Use the Number component instead.
  warnings.warn(value)
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
Exception in thread Thread-3 (gentask):
Traceback (most recent call last):
  File "C:\Users\Standard\miniconda3\envs\textgen\lib\threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "C:\Users\Standard\miniconda3\envs\textgen\lib\threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\Standard\text-generation-webui\modules\callbacks.py", line 63, in gentask
    ret = self.mfunc(callback=_callback, **self.kwargs)
  File "C:\Users\Standard\text-generation-webui\modules\text_generation.py", line 218, in generate_with_callback
    shared.model.generate(**kwargs)
  File "C:\Users\Standard\miniconda3\envs\textgen\lib\site-packages\torch\autograd\grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "C:\Users\Standard\miniconda3\envs\textgen\lib\site-packages\transformers\generation\utils.py", line 1452, in generate
    return self.sample(
  File "C:\Users\Standard\miniconda3\envs\textgen\lib\site-packages\transformers\generation\utils.py", line 2468, in sample
    outputs = self(
  File "C:\Users\Standard\miniconda3\envs\textgen\lib\site-packages\torch\nn\modules\module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "C:\Users\Standard\miniconda3\envs\textgen\lib\site-packages\transformers\models\llama\modeling_llama.py", line 765, in forward
    outputs = self.model(
  File "C:\Users\Standard\miniconda3\envs\textgen\lib\site-packages\torch\nn\modules\module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "C:\Users\Standard\text-generation-webui\repositories\GPTQ-for-LLaMa\llama_inference_offload.py", line 146, in forward
    layer_outputs = decoder_layer(
  File "C:\Users\Standard\miniconda3\envs\textgen\lib\site-packages\torch\nn\modules\module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "C:\Users\Standard\miniconda3\envs\textgen\lib\site-packages\transformers\models\llama\modeling_llama.py", line 309, in forward
    hidden_states, self_attn_weights, present_key_value = self.self_attn(
  File "C:\Users\Standard\miniconda3\envs\textgen\lib\site-packages\torch\nn\modules\module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "C:\Users\Standard\miniconda3\envs\textgen\lib\site-packages\transformers\models\llama\modeling_llama.py", line 209, in forward
    query_states = self.q_proj(hidden_states).view(bsz, q_len, self.num_heads, self.head_dim).transpose(1, 2)
  File "C:\Users\Standard\miniconda3\envs\textgen\lib\site-packages\torch\nn\modules\module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "C:\Users\Standard\text-generation-webui\repositories\GPTQ-for-LLaMa\quant.py", line 278, in forward
    quant_cuda.vecquant4matmul(x, self.qweight, y, self.scales, self.qzeros, self.groupsize)
TypeError: vecquant4matmul(): incompatible function arguments. The following argument types are supported:
    1. (arg0: at::Tensor, arg1: at::Tensor, arg2: at::Tensor, arg3: at::Tensor, arg4: at::Tensor) -> None

Invoked with: tensor([[ 0.0436, -0.0149,  0.0150,  ...,  0.0267,  0.0112, -0.0011],
        [-0.0108,  0.0345, -0.0282,  ...,  0.0073, -0.0096,  0.0204],
        [-0.0362, -0.0222, -0.0107,  ..., -0.0035, -0.0108,  0.0189],
        ...,
        [ 0.0324,  0.0055,  0.0122,  ...,  0.0099, -0.0175,  0.0141],
        [ 0.0160, -0.0103, -0.0197,  ...,  0.0249, -0.0164,  0.0180],
        [-0.0431, -0.0260,  0.0012,  ...,  0.0075, -0.0076, -0.0037]],
       device='cuda:0'), tensor([[-1110991750,  2069465943,   910865574,  ..., -1740800632,
          -699816361,  1702373496],
        [-1381311558, -1777805958,   844506949,  ...,  1188386441,
          1768455033,  1702187603],
        [ -308546134, -1519745622,  1479621281,  ..., -1195758162,
         -1282725701, -1671187898],
        ...,
        [-1231525415, -1763260567, -2020702810,  ...,  2019146646,
          1214946132, -1485408886],
        [-1765312474, -1466198217,  1517197483,  ...,   143096486,
          1667650470,  1768131158],
        [  952884391,  1808641601,  -966114168,  ..., -1198823032,
          1700290168,  1135113064]], device='cuda:0', dtype=torch.int32), tensor([[0., 0., 0.,  ..., 0., 0., 0.],
        [0., 0., 0.,  ..., 0., 0., 0.],
        [0., 0., 0.,  ..., 0., 0., 0.],
        ...,
        [0., 0., 0.,  ..., 0., 0., 0.],
        [0., 0., 0.,  ..., 0., 0., 0.],
        [0., 0., 0.,  ..., 0., 0., 0.]], device='cuda:0'), tensor([[0.0071, 0.0038, 0.0034,  ..., 0.0082, 0.0083, 0.0102],
        [0.0062, 0.0039, 0.0039,  ..., 0.0091, 0.0089, 0.0103],
        [0.0263, 0.0128, 0.0094,  ..., 0.0146, 0.0180, 0.0105],
        ...,
        [0.0085, 0.0046, 0.0042,  ..., 0.0101, 0.0107, 0.0120],
        [0.0070, 0.0043, 0.0038,  ..., 0.0087, 0.0073, 0.0114],
        [0.0069, 0.0035, 0.0041,  ..., 0.0115, 0.0081, 0.0117]],
       device='cuda:0'), tensor([[ 1987474810, -2057993112,  1449691205,  ...,  1465415557,
          2018924135,  1468491637],
        [ 1988589413,  2005358472,  1718179430,  ...,  1719096918,
         -2021099674,  1716946790],
        [-2042324907,  1485396104, -1807136887,  ...,  1750685016,
          1988515688,  1969702484],
        ...,
        [ 1969649526,  1484281719,  1717004424,  ..., -1989712328,
          2021095014,  1952929382],
        [ 2003199864, -1772587433,  2020120166,  ...,  1716942728,
          1701209734,  1449625206],
        [ 1970694008, -1806407580, -2022270861,  ..., -1754892441,
          1986418039,  1720088424]], device='cuda:0', dtype=torch.int32), 128
```

### System Info

```shell
Windows 10
NVIDIA GeForce GTX 1050 Ti
```

**Comment by jllllll** [comment]
There was a bug in GPTQ-for-LLaMa. It was fixed about an hour after you downloaded it. Updating should fix this.

**Comment by whitepapercg** [comment]
> There was a bug in GPTQ-for-LLaMa. It was fixed about an hour after you downloaded it. Updating should fix this.

git clone https://github.com/qwopqwop200/GPTQ-for-LLaMa

or another command?

**Comment by Ph0rk0z** [comment]
you can delete it and re clone it or do a git pull inside the folder.

**Comment by oobabooga** [comment]
Note that the old 4-bit weights are not compatible anymore. You must download the updated weights.

The guide has been updated. If you follow all the steps, it should work: https://github.com/oobabooga/text-generation-webui/wiki/LLaMA-model#installation

**Comment by whitepapercg** [comment]
> Note that the old 4-bit weights are not compatible anymore. You must download the updated weights.
> 
> The guide has been updated. If you follow all the steps, it should work: https://github.com/oobabooga/text-generation-webui/wiki/LLaMA-model#installation

I get the same error after doing everything.
We need current 4bit support for windows without WSL, this worked on earlier versions of the package, but broke with the update

**Comment by BugMonkey42335** [comment]
After doing a clean install using jllllll one-click-installer I was able to get llama-7b-4bit up and running.

https://github.com/jllllll/one-click-installers

**Comment by brandonj60** [comment]
> After doing a clean install using jllllll one-click-installer I was able to get llama-7b-4bit up and running.
> 
> https://github.com/jllllll/one-click-installers

What version of C++ build tools do you have installed? The one click installer fails compiling during setup_cuda install for me even though I have VS 2019 16.9.6.

**Comment by jllllll** [comment]
> > After doing a clean install using jllllll one-click-installer I was able to get llama-7b-4bit up and running.
> > https://github.com/jllllll/one-click-installers
> 
> I'm getting the same error you were. What version of C++ build tools do you have installed? The one click installer fails compiling during setup_cuda install for me even though I have VS 2019 16.9.6.

I use VS 2022 latest version. If you can't get it to compile, try updating VS, restart your computer, and then reinstall the webui (new virtual environment). Windows seems to be pretty temperamental with compiling, I tried all those things and it eventually just started working. If it comes to it, you can use the one-click-installer mentioned above and it should install a precompiled wheel when the compilation fails.

**Comment by 8WSR0hX** [comment]
Make sure that you do not use the --groupsize flag with models that were not quantized with groupsize.

**Comment by BugMonkey42335** [comment]
> > After doing a clean install using jllllll one-click-installer I was able to get llama-7b-4bit up and running.
> > https://github.com/jllllll/one-click-installers
> 
> I'm getting the same error you were. What version of C++ build tools do you have installed? The one click installer fails compiling during setup_cuda install for me even though I have VS 2019 16.9.6.

I let the installer use the precompiled wheel, I don't have C++ build tools installed.

**Comment by DiXiaoO** [comment]
I have the same error when trying to generate a response

I have 1660 Super and I installed CUDA 11.7 and torch which fits this version
I also have VS 2019
I tried to reinstall as suggested here but it didn't work.

**Comment by Noospherix** [comment]
I had the same issue. Fixed it with this: https://github.com/oobabooga/text-generation-webui/issues/416#issuecomment-1475078571. I already had done everything except removing and running the setup for GPTQ-for-LLaMa again. It was not enough to pull the latest updates I had to remove it, clone it again, and run 'python setup_cuda.py install' again. Also as others have noted if you remove everything and start over it will reinstall this and should fix the issue.
